data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/network_action/PstSetpoint.java

```diff
@@ -39,6 +41,11 @@ public void setSetpoint(double setpoint) {
 
     @Override
     public void apply(Network network) {
-        throw new UnsupportedOperationException();
+        PhaseTapChanger phaseTapChanger = network.getTwoWindingsTransformer(getNetworkElement().getId()).getPhaseTapChanger();
+        if (phaseTapChanger.getHighTapPosition() >= setpoint && phaseTapChanger.getLowTapPosition() <= setpoint) {
+            phaseTapChanger.setTapPosition((int) setpoint);
+        } else {
+            throw new FaraoException("PST cannot be set because setpoint is out of PST boundaries");
+        }
```

Comment:
Maybe add a specific exception if PST does not exist in network

Reply:
I added a specific message.

You'll see that now the implementation of PstSetpoint use PstRange which is a bit weird but avoids code duplication. But it might be due to a bad conception here, we could check that later.